Sources/OuterCore/Schedule.swift

```diff
@@ -0,0 +1,168 @@
+/// Schedule.swift
+///
+/// Copyright 2017, The Silt Language Project.
+///
+/// This project is released under the MIT license, a copy of which is
+/// available in the repository.
+
+/// A schedule adds SSA-like operations before a CPS-style call. 
+final class Schedule {
+  /// Whether this operation is scheduled early (before the 'terminator'), or
+  /// late (after the 'terminator').
+  enum Tag {
+    case early
+    case late
+  }
+
+  /// A 'block' abstraction that 
+  final class Block: Equatable, Hashable {
+    var parent: Continuation
+    var primops: [PrimOp]
+    var index: Int
+
+    init(_ parent: Continuation, _ primops: [PrimOp], _ idx: Int) {
+      self.parent = parent
+      self.primops = primops
+      self.index = idx
+    }
+
+    public static func ==(lhs: Block, rhs: Block) -> Bool {
+      return lhs === rhs
+    }
+    
+    public var hashValue: Int {
+      return "\(ObjectIdentifier(self).hashValue)".hashValue
+    }
+  }
+
+  let scope: Scope
+  let tag: Tag
+  var blocks: [Block] = []
+  var indices: [Continuation: Int] = [:]
+  init(_ scope: Scope, _ tag: Tag) {
+    self.scope = scope
+    self.tag = tag
+
+    // until we have sth better simply use the RPO of the CFG
+    var i = 0
+    for n in scope.cfg.reversePostOrder {
+      defer { i += 1 }
+      self.blocks.append(Block(n, [], i))
+      self.indices[n] = i
+    }
+    _ = Scheduler(self)
+  }
+
+  func block(_ c : Continuation) -> Block {
+    return self.blocks[self.indices[c]!]
+  }
+
+  func dump() {
+    for block in self.blocks {
+      block.parent.dump()
+      for primop in block.primops {
+        primop.dump()
+      }
+    }
+  }
+}
+
+private final class Scheduler {
+  let scope: Scope
+  let schedule: Schedule
+  private var def2uses = [Value: [Operand]]()
+
+  private var def2early = [Value: Continuation]()
+  private var def2late = [Value: Continuation]()
+
+  init(_ schedule: Schedule) {
+    self.scope = schedule.scope
+    self.schedule = schedule
+    self.computeDef2Uses()
+    switch schedule.tag {
+    case .early:
+      for (_, uses) in self.def2uses {
```

Comment:
Iterating a dictionary leads to unstable code ordering. You can build a dominator tree to establish indices where continuations should reside in the Block